src/Products/Conversion/Manager/ConversionManager.cs

```diff
@@ -1,57 +1,82 @@
         {
             string sourceType = Path.GetExtension(postedData.guid).TrimStart('.');
             string destinationType = postedData.GetDestinationType();
-            string resultFileName = Path.GetFileNameWithoutExtension(postedData.guid) + "." + postedData.GetDestinationType();
-            dynamic saveOptions = GetSaveOptions(sourceType, destinationType);
+            string documentGuid = postedData.guid;
+            string filesDirectory = globalConfiguration.GetConversionConfiguration().GetResultDirectory();
+            string outputFile = Path.Combine(filesDirectory, Path.GetFileNameWithoutExtension(documentGuid) + "." + postedData.GetDestinationType());
+            string destDocumentType = supportedImageFormats.Contains(Path.GetExtension("." + destinationType)) ? "image" : postedData.GetDestDocumentType();
+            string fileNameWoExt = Path.GetFileNameWithoutExtension(postedData.guid);
 
-            ConvertedDocument convertedDocument = conversionHandler.Convert(postedData.guid, saveOptions);
-
-            if (convertedDocument.PageCount > 1 && saveOptions is ImageSaveOptions)
+            using (Converter converter = new Converter(postedData.guid))
             {
-                for (int i = 1; i <= convertedDocument.PageCount; i++)
+                var convertOptions = GetConvertOptions(destDocumentType, destinationType);
+
+                var documentInfo = converter.GetDocumentInfo();
+
+                if (convertOptions is ImageConvertOptions)
                 {
-                    string fileName = Path.GetFileNameWithoutExtension(resultFileName) + "-page" + i + "." + Path.GetExtension(resultFileName);
-                    convertedDocument.Save(fileName, i);
+                    string outputFileTemplate = Path.Combine(filesDirectory, fileNameWoExt + "-{0}." + destinationType);
+
+                    if ((documentInfo is SpreadsheetDocumentInfo && ((SpreadsheetDocumentInfo)documentInfo).WorksheetsCount == 1)
+                        || documentInfo.PagesCount == 1)
+                    {
+                        outputFileTemplate = Path.Combine(filesDirectory, fileNameWoExt + "." + destinationType);
+                    }
+
+                    SavePageStream getPageStream = page => new FileStream(string.Format(outputFileTemplate, page), FileMode.Create);
+                    converter.Convert(getPageStream, convertOptions);
+                }
+                else
+                {
+                    converter.Convert(outputFile, convertOptions);
                 }
-            }
-            else
-            {
-                convertedDocument.Save(resultFileName);
             }
         }
 
-        private SaveOptions GetSaveOptions(string sourceType, string destinationType)
+        private ConvertOptions GetConvertOptions(string destDocumentType, string destinationType)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make 'GetConvertOptions' a static method.](https://app.codacy.com/gh/groupdocs-conversion/GroupDocs.Conversion-for-.NET-MVC/pullRequest?prid=5332801)